std/compress/lzss/compress.go

```diff
@@ -0,0 +1,218 @@
+package lzss
+
+import (
+	"bytes"
+	"fmt"
+	"math/bits"
+
+	"github.com/consensys/gnark/std/compress/lzss/suffixarray"
+	"github.com/icza/bitio"
+)
+
+type Compressor struct {
+	// end int
+	buf bytes.Buffer
+	bw  *bitio.Writer
+
+	inputIndex *suffixarray.Index
+	inputSa    [maxInputSize]int32 // suffix array space.
+
+	dictData  []byte
+	dictIndex *suffixarray.Index
+	dictSa    [maxDictSize]int32 // suffix array space.
+}
+
+type CompressionMode uint8
+
+const (
```

Comment:
need doc to understand difference between "BestCompression" and "BestSnarkCompression" for example

Reply:
That's exactly why I think it's better to give it a name like `WordLength` that actually denotes its function, and then explain in docs that lower values increase the compression ratio and higher value decrease the decompressor's number of constraints.